tangelo/toolboxes/molecular_computation/mm_charges_solver.py

```diff
@@ -0,0 +1,161 @@
+
+    Returns:
+        List[str]: The list of pdb equivalent filenames.
+    """
+
+    try:
+        import openbabel
+        from openbabel.openbabel import obErrorLog
+        obErrorLog.SetOutputLevel(0)
+    except ModuleNotFoundError:
+        raise ModuleNotFoundError(f"openbabel is required to use get_charges_and_coords_from_pdb_file when supplying only a pdb file."
+                                  "install with 'pip install openbabel-wheel'")
+
+    filesold = [filenames] if isinstance(filenames, str) else filenames
+
+    mol = openbabel.openbabel.OBMol()
+    conv = openbabel.openbabel.OBConversion()
+
+    files = []
+    for i, file in enumerate(filesold):
+        if file[-3:] != "pdb":
+            mol = openbabel.openbabel.OBMol()
+            conv.SetOutFormat("pdb")
+            conv.ReadFile(mol, file)
+            conv.WriteFile(mol, f"tmp{i}.pdb")
+            files += ["tmp{0}.pdb"]
+        else:
+            files += [file]
+
+    return files
+
+
+def get_default_mm_package() -> Type[MMChargesSolver]:
+    if is_package_installed("rdkit"):
+        return MMChargesSolverOpenMM()
+    elif is_package_installed("openmm"):
+        return MMChargesSolverRDKit()
+    else:
+        return None
+
+
+def get_mm_package(package: str = "default"):
+    """Return the MMCharges instance
+
+    Args:
+        package (str): The MM backend to use
+
+    Returns:
+        Type[MMChargesSolver]: The MMChargesSolver that computes the partial charges for given geometries
+    """
+    if package == "default":
+        return get_default_mm_package()
+    elif package.lower() == "openmm":
+        return MMChargesSolverOpenMM()
+    elif package.lower() == "rdkit":
+        return MMChargesSolverRDKit()
+    else:
+        raise ValueError(f"Requested MM package of {package} is not currently supported in Tangelo")
+
+
+class MMChargesSolverOpenMM(MMChargesSolver):
+    def __init__(self, force_field_params=('amber14-all.xml', 'amber14/tip3pfb.xml')):
+        self.force_field_params = force_field_params
+        pass
+
+    def get_charges(self, files) -> List[Tuple[float, Tuple[float, float, float]]]:
+        from openmm.app.pdbfile import PDBFile
+        from openmm.app.forcefield import ForceField
+        from openmm import NonbondedForce
+        from openmm.app import Modeller
+        import openbabel
+
+        pdbfiles = convert_files_to_pdbs(files)
+
+        pdbs = [PDBFile(file) for file in pdbfiles]
+        modeller = Modeller(pdbs[0].topology, pdbs[0].positions)
+        for pdb in pdbs[1:]:
+            modeller.add(pdb.topology, pdb.positions)
+        forcefield = ForceField(*self.force_field_params)
+        system = forcefield.createSystem(modeller.topology)
+        nonbonded = [f for f in system.getForces() if isinstance(f, NonbondedForce)][0]
+        charges = [nonbonded.getParticleParameters(i)[0]._value for i in range(system.getNumParticles())]
+        # Obtain xyz string
```

Comment:
Don't say "xyz string" but something that makes chemical sense. Did you mean "Generate xyz coordinates of atoms" ? (is it clear what those atoms are from context btw?)